packages/react-core/src/components/SearchInput/SearchInput.tsx

```diff
@@ -277,61 +300,146 @@ const SearchInputBase: React.FunctionComponent<SearchInputProps> = ({
+          <span>
+            <AdvancedSearchMenu
+              value={value}
+              parentRef={searchInputRef}
+              parentInputRef={searchInputInputRef}
+              onSearch={onSearch}
+              onClear={onClear}
+              onChange={onChange}
+              onToggleAdvancedMenu={onToggle}
+              resetButtonLabel={resetButtonLabel}
+              submitSearchButtonLabel={submitSearchButtonLabel}
+              attributes={attributes}
+              formAdditionalItems={formAdditionalItems}
+              hasWordsAttrLabel={hasWordsAttrLabel}
+              advancedSearchDelimiter={advancedSearchDelimiter}
+              getAttrValueMap={getAttrValueMap}
+              isSearchMenuOpen={isSearchMenuOpen}
+            />
+          </span>
+        );
 
-      const AdvancedSearchWithPopper = (
-        <div className={css(className)} ref={searchInputRef} {...props}>
-          <Popper
-            trigger={buildSearchTextInputGroupWithExtraButtons()}
-            popper={AdvancedSearch}
-            isVisible={isSearchMenuOpen}
-            enableFlip={true}
-            appendTo={() => appendTo || searchInputRef.current}
-          />
-        </div>
-      );
+        const AdvancedSearchWithPopper = (
+          <div className={css(className)} ref={searchInputRef} {...props}>
+            <Popper
+              trigger={buildSearchTextInputGroupWithExtraButtons()}
+              popper={AdvancedSearch}
+              isVisible={isSearchMenuOpen}
+              enableFlip={true}
+              appendTo={() => appendTo || searchInputRef.current}
+            />
+          </div>
+        );
 
-      const AdvancedSearchInline = (
-        <div className={css(className)} ref={searchInputRef} {...props}>
-          {buildSearchTextInputGroupWithExtraButtons()}
-          {AdvancedSearch}
-        </div>
-      );
+        const AdvancedSearchInline = (
+          <div className={css(className)} ref={searchInputRef} {...props}>
+            {buildSearchTextInputGroupWithExtraButtons()}
+            {AdvancedSearch}
+          </div>
+        );
 
-      return appendTo !== 'inline' ? AdvancedSearchWithPopper : AdvancedSearchInline;
+        return appendTo !== 'inline' ? AdvancedSearchWithPopper : AdvancedSearchInline;
+      }
+      buildSearchTextInputGroupWithExtraButtonsAndExpand({ ...searchInputProps });
     }
+    return buildSearchTextInputGroupWithExpand({ ...searchInputProps });
+  } else if (!!onToggleExpand && !isSearchInputExpanded) {
+    return (
+      <InputGroup {...searchInputProps}>
+        <Button
+          variant={ButtonVariant.plain}
+          aria-label={expandedLabel()}
+          icon={<SearchIcon />}
+          onClick={onExpandHandler}
+        ></Button>
+      </InputGroup>
+    );
+  } else {
+    if (!!onSearch || attributes.length > 0 || !!onToggleAdvancedSearch) {
+      if (attributes.length > 0) {
+        const AdvancedSearch = (
+          <span>
```

Comment:
Is this code simply repeated from the code around lines 333? If so, we should avoid duplicate code as much as possible.